pennylane/drawer/tape_text.py

```diff
@@ -49,6 +49,10 @@ def _add_op(op, layer_str, wire_map, decimals, cache):
         if w not in control_wires:
             layer_str[wire_map[w]] += label
 
+    if len(op.wires) == 0:  # operation (e.g. barrier, snapshot) across all wires
+        for i, s in enumerate(layer_str):
+            layer_str[i] = s + label
```

Comment:
```suggestion
    if len(op.wires) == 0:  # operation (e.g. barrier, snapshot) across all wires
        for i, s in enumerate(layer_str):
            layer_str[i] = s + label
    else:
        for w in op.wires:
           if w not in control_wires:
               layer_str[wire_map[w]] += label
```
Exactly right to get the job done.

We could combine this with lines 48-50 though.  Indentation on suggestion is horrible because of writing in here, but you should get the gist.

Reply:
Gotcha, thanks :)